Add cpu/gloo tests for sharded tensor distributed checkpoint

zzzwen opened the pull request:
SharedTensor checkpoint does not depend on NCCL, replicate the GPU cases for CPU and enable development on mac

wanchaol commented on test/distributed/_shard/checkpoint/test_file_system_checkpoint_cpu.py:
```diff
@@ -0,0 +1,542 @@
+    )
+    sys.exit(0)
+
+def _sharded_tensor_gather(
```
https://github.com/pytorch/pytorch/blob/394aca9a26cd760d1f761821452b0d308ac35b03/torch/distributed/_shard/sharded_tensor/api.py#L269

This method already been merged by @rodrigobdz so I guess we can safely remove this.

wanchaol reviewed (approved):
Looks good! Left one small comment, please make sure CI pass before landing.

zzzwen committed:
Add cpu/gloo tests for sharded tensor distributed checkpoint

SharedTensor checkpoint does not depend on NCCL, replicate the GPU cases for CPU and enable development on mac

zzzwen commented:
@pytorchbot merge
